### PR TITLE
test new AMI

### LINKED ISSUE #16265: The action runner EC2 image used for connector publication may not have enough resource
## Summary
Running integration tests `source-mysql` and `source-mysql-strict-encrypt` by the `/test` command completes without any problem, while running the `/publish` command for the same two connectors would stuck there for hours.

The most recent example is here: https://github.com/airbytehq/airbyte/pull/16259

One difference between the `/test` and `/publish` command is that the former uses EC2 image `ami-0d648081937c75a73` with 80GB disk, while the latter uses the default `ami-0f23be2f917510c26`. I suspect the latter does not have enough disk space or memory to run the integration tests. When I tried to use `ami-0d648081937c75a73` to run the `/publish` command, it failed because there is no `docker buildx` in that image.

Could we update `ami-0d648081937c75a73` to have a newer version of docker that supports `docker buildx`, or add more resource to the default `ami-0f23be2f917510c26`?

Here is a summary:

| | `/test` | `/publish` |
| --- | --- | --- |
| EC2 image | `ami-0d648081937c75a73` | `ami-0f23be2f917510c26` |
| What it does | Run integration test. | Run integration test + publish docker image. |
| Problem | Does not have `docker buildx`. | May not have enough resource (e.g. disk). |
| Request (either one) | Install a newer version of `docker` with `docker buildx` in this image. | Keep the dependencies. Add more resource to this image. |

### PR DESCRIPTION
**Pull request by git-phu**
Use new github runner AMI that is fully documented here: https://github.com/airbytehq/airbyte-cloud/pull/3445

closes https://github.com/airbytehq/airbyte/issues/16265

### CODE REVIEW

File: .github/actions/start-aws-runner/action.yml [hunk 1]
```diff
@@ -8,8 +8,8 @@ inputs:
   github-token:
     required: true
   ec2-image-id:
-    # github-self-hosted-runner-ubuntu-20-100g-disk-with-cypress-deps
-    default: "ami-005924fb76f7477ce"
+    # https://internal-docs.airbyte.io/Things-To-Know/Build-Runner-Images#ami-05b3422b3649a6911
```
Comment (git-phu) [inline .github/actions/start-aws-runner/action.yml:11]: should we include a link to internal docs in oss?

Comment (evantahler) [reply .github/actions/start-aws-runner/action.yml:11]: I think that's fine

**Comment by evantahler** [comment]
Can you remove the AMI override here (https://github.com/airbytehq/airbyte/blob/0dddc1e647de9fb3f6d03c09e99429a58d54c676/.github/workflows/test-command.yml#L65) and then we can test out a `/test` connector run as well?

**Comment by git-phu** [comment]
> Can you remove the AMI override here (
> 
> https://github.com/airbytehq/airbyte/blob/0dddc1e647de9fb3f6d03c09e99429a58d54c676/.github/workflows/test-command.yml#L65
> 
> ) and then we can test out a `/test` connector run as well?

@evantahler done!

**Comment by evantahler** [comment]
This PR probably closes https://github.com/airbytehq/airbyte/issues/16265

**Review by evantahler — APPROVED** [review]
YOLO since we can't test it I guess 🤷

**Comment by git-phu** [comment]
gonna merge this now and watch for regressions

**Pull request merged**